src/catalog/src/durable/impls/persist.rs

```diff
@@ -121,14 +129,19 @@ impl UnopenedPersistCatalogState {
             )
             .await
             .expect("invalid usage");
-        UnopenedPersistCatalogState {
+        let mut handle = UnopenedPersistCatalogState {
             write_handle,
             read_handle,
             persist_client,
             shard_id,
             snapshot_cache: None,
             metrics,
+            epoch: MIN_EPOCH,
+        };
+        if let Some(epoch) = handle.get_current_epoch().await {
```

Comment:
If this returns `None`, why is it correct to leave `handle.epoch` as the default  `MIN_EPOCH`? `MIN_EPOCH` is the logical equivalent of `None` because we will never write epoch 1 down since we always increment it to at least 2?

Reply:
If that's the case, could we invert the logic and do like `epoch = hadnle.get_current_epoch().unwrap_or_default(MIN_EPOCH)` and then make `handle` not `mut` and directly set that field?

Reply:
> If this returns None, why is it correct to leave handle.epoch as the default MIN_EPOCH? MIN_EPOCH is the logical equivalent of None because we will never write epoch 1 down since we always increment it to at least 2?

Hm, that's a good point. It does end up working out because of exactly what you said, but it's probably better to explicitly set `handle.epoch` to `None`.

> If that's the case, could we invert the logic and do like epoch = hadnle.get_current_epoch().unwrap_or_default(MIN_EPOCH) and then make handle not mut and directly set that field?

We can't do exactly that because we have a bit of a chicken and egg problem. We can't call `handle.get_current_epoch()` to get the epoch until we initialize the `handle`, but we can't initialize the `handle` until we have the epoch. I'll do some refactoring so the method is not part of the `handle`.

Reply:
Ok, I pushed a small refactor for this.